### PR TITLE
feat: add user ability to configure max write #

### PR DESCRIPTION
**Pull request by salesh**
Related to https://github.com/pinojs/sonic-boom/issues/138 

I hope I covered everything - please let me know if I need to add/change something.

### CODE REVIEW

**Review by jsumners — APPROVED** [review]
Looks good to me.

File: types/index.d.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export type SonicBoomOpts = {
     dest?: string | number
     maxLength?: number
     minLength?: number
+    maxWrite?: number
```
Comment (mcollina) [inline types/index.d.ts:14]: We actually test our types with tsd! Could you add an assertion for this?

Comment (salesh) [reply types/index.d.ts:14]: @mcollina please correct me if I am wrong - by checking out how **tsd** works and checking out some other pino repositories I guess I only need to make change like this https://github.com/pinojs/sonic-boom/pull/142/commits/42799c6ce275afd76f447a7d57b0ebf075618605


Let me know if my understanding is bad, and thanks for sharing with me **tsd**

**Review by mcollina — APPROVED** [review]
lgtm

File: types/index.d.ts (hunk 1 above)
Comment (mcollina) [reply types/index.d.ts:14]: that's ok!

**Pull request merged**